src/services/blockchain/__tests__/eth-bc-service.test.ts

```diff
@@ -237,38 +247,91 @@ describe('ETH service with mock wallet',  () => {
     const nonce = 5
     const gasPrice = BigNumber.from(1000)
     const gasEstimate = BigNumber.from(10*1000)
+    const txResponse = {foo: 'bar'}
 
     provider.getBalance.mockReturnValue(balance)
     provider.getTransactionCount.mockReturnValue(nonce)
     provider.getGasPrice.mockReturnValue(gasPrice)
     provider.estimateGas.mockReturnValue(gasEstimate)
 
     const mockTrySendTransaction = jest.fn()
+    const mockConfirmTransactionSuccess = jest.fn()
     ethBc._trySendTransaction = mockTrySendTransaction
-    mockTrySendTransaction.mockRejectedValue({code: ErrorCode.TIMEOUT})
+    ethBc._confirmTransactionSuccess = mockConfirmTransactionSuccess
+    mockTrySendTransaction.mockReturnValue(txResponse)
+    mockConfirmTransactionSuccess.mockRejectedValue({code: ErrorCode.TIMEOUT})
 
     const cid = new CID('bafyreic5p7grucmzx363ayxgoywb6d4qf5zjxgbqjixpkokbf5jtmdj5ni');
     await expect(ethBc.sendTransaction(cid)).rejects.toThrow("Failed to send transaction")
 
     expect(mockTrySendTransaction).toHaveBeenCalledTimes(3)
 
-    const [txData0, attemptNum0, network0, transactionTimeoutSecs0] = mockTrySendTransaction.mock.calls[0]
+    const [txData0, attemptNum0, network0] = mockTrySendTransaction.mock.calls[0]
     expect(attemptNum0).toEqual(0)
     expect(network0).toEqual(config.blockchain.connectors.ethereum.network)
-    expect(transactionTimeoutSecs0).toEqual(config.blockchain.connectors.ethereum.transactionTimeoutSecs)
     expect(txData0).toMatchSnapshot()
 
-    const [txData1, attemptNum1, network1, transactionTimeoutSecs1] = mockTrySendTransaction.mock.calls[1]
+    const [txData1, attemptNum1, network1] = mockTrySendTransaction.mock.calls[1]
     expect(attemptNum1).toEqual(1)
     expect(network1).toEqual(config.blockchain.connectors.ethereum.network)
-    expect(transactionTimeoutSecs1).toEqual(config.blockchain.connectors.ethereum.transactionTimeoutSecs)
     expect(txData1).toMatchSnapshot()
 
-    const [txData2, attemptNum2, network2, transactionTimeoutSecs2] = mockTrySendTransaction.mock.calls[2]
+    const [txData2, attemptNum2, network2] = mockTrySendTransaction.mock.calls[2]
     expect(attemptNum2).toEqual(2)
     expect(network2).toEqual(config.blockchain.connectors.ethereum.network)
-    expect(transactionTimeoutSecs2).toEqual(config.blockchain.connectors.ethereum.transactionTimeoutSecs)
     expect(txData2).toMatchSnapshot()
+
+    expect(mockConfirmTransactionSuccess).toHaveBeenCalledTimes(3)
+
+    const [txResponse0, network3, transactionTimeoutSecs0] = mockConfirmTransactionSuccess.mock.calls[0]
+    expect(network3).toEqual(config.blockchain.connectors.ethereum.network)
```

Comment:
Why do we need to check the network in every result for this test?

Reply:
Yeah, I guess I was just being thorough but you're right it's probably overkill and making the test unnecessarily brittle.  Updated